tests/run/fused_cpdef.pyx

```diff
@@ -32,23 +48,64 @@ def test_fused_cpdef():
     A, x is long 2 long
     A, x is long 2 long
     A, x is long 2 long
+    <BLANKLINE>
+    B, x is long 2 long
     """
     func1[int](None, 2)
     func1[long](None, 2)
     func1(None, 2)
 
-    print
+    print()
 
     pyfunc[cy.int](None, 2)
     pyfunc(None, 2)
 
-    print
+    print()
 
     A.meth[cy.int](A(), 2)
     A.meth(A(), 2)
     A().meth[cy.long](2)
     A().meth(2)
 
+    print()
+
+    B().meth(2)
+
+
+midimport_run = io.StringIO()
+if sys.version_info.major < 3:
+    # Monkey-patch midimport_run.write to accept non-unicode strings under Python 2.
+    midimport_run.write = lambda c: io.StringIO.write(midimport_run, unicode(c))
+
+realstdout = sys.stdout
+sys.stdout = midimport_run
+
+try:
+    test_fused_cpdef()
```

Comment:
This runs the test function at module import time, rather than test time. What's the reason for that?

Reply:
In one of the intermediate versions where I was keeping the signature indices in the module namespace, dynamically calling a `fused` `cpdef` function during module import would fail because the signature indices wouldn't be initialized until a later point in the import.

It's no longer such an issue now that the signature indices are back in default keyword arguments, but this test is meant to prevent that behaviour from potentially becoming a problem again with future changes.

The docstring for `test_fused_cpdef()` is still there, so it also gets run at test time. `test_midimport_run()` echoes the result saved from running it at module import time to make sure that works too.